### PR TITLE
[TSI] Add more details in readme

### PR DESCRIPTION
**Pull request by bikamani**
Add missing details in readme.

### CODE REVIEW

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md [hunk 1]
```diff
@@ -78,7 +84,33 @@ Assembly properties required for running unit tests.
 
 ## Troubleshooting
 
-Coming soon.
+All Time Series Insights service operations will contain helpful error information in response [TimeSeriesOperationError](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Generated/Models/TimeSeriesOperationError.cs).
```
Comment (bikamani) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md:87]: @barustum Does this look right?

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md [hunk 2]
```diff
@@ -25,7 +25,13 @@ In order to interact with the Azure Time Series Insights service, you will need
 
 ## Key concepts
 
-Coming soon.
+The main concepts of Time Series Insights client include:
+
+Instances client: To perform operations such as creating, listing, replacing and deleting Time Series instances.
```
Comment (ericmitt) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md:30]: This doesn't format as a list in md

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md [hunk 3]
```diff
@@ -25,7 +25,13 @@ In order to interact with the Azure Time Series Insights service, you will need
...
+Types client: To perform operations such as creating, listing, replacing and deleting Time Series types.
+Hierarchies client: To perform operations such as creating, listing, replacing and deleting Time Series hierarchies.
+Model Settings client: To perform operations such as getting and updating Time Series Model configuration settings.
+Query client: To query for events, series and aggregate series on Time Series Insights.
 
 ### Thread safety
 We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-methods-thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
```
Comment (ericmitt) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md:37]: Unclear, perhaps better this : This ensures that reusing client instances is always safe, even across threads.

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md (hunk 1 above)
Comment (barustum) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md:87]: I wouldn't say 'All'. Reason being, some APIs do not return error or result. For example, GetAsync for Hierarchies client return TimeSeriesHierarchy. 

Maybe something along these lines:
Time Series Insights service operation failures are usually returned to the user as [TimeSeriesOperationError](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Generated/Models/TimeSeriesOperationError.cs). The TimeSeriesOperationError response is either returned directly by the client library API, or as a nested property within the actual response for the client library API. For example, the DeleteByName API that is part of the hierarchies client returns a TimeSeriesOperationError directly. Whereas, the Replace API that is part of the instances client returns a InstancesOperationResult, which has a TimeSeriesOperationError property nested within it.

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md (hunk 2 above)
Comment (bikamani) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/README.md:30]: will do.

**Review by barustum — APPROVED** [review]

**Pull request merged**